spec/02-integration/11-dbless/03-config_persistence_spec.lua

```diff
@@ -124,31 +124,38 @@ describe("dbless persistence with a declarative config #off", function()
         database = "off",
         declarative_config = yaml_file,
     }))
-    local proxy_client = assert(helpers.proxy_client())
-    local res = assert(proxy_client:get("/test", { headers = { host = "example1.dev" } }))
-    assert.res_status(401, res)
 
-    res = assert(proxy_client:get("/500", { headers = { host = "example1.dev" } }))
-    assert.res_status(404, res) -- 404, only the declarative config is loaded
-    proxy_client:close()
+    assert
+      .with_timeout(15)
+      .eventually(function ()
+        local proxy_client = assert(helpers.proxy_client())
+        local res = assert(proxy_client:get("/test", { headers = { host = "example1.dev" } }))
+        assert.res_status(401, res) -- 401, should load the declarative config
+
+        res = assert(proxy_client:get("/500", { headers = { host = "example1.dev" } }))
+        assert.res_status(404, res) -- 404, should not load the persisted lmdb config
+
+        proxy_client:close()
+      end)
   end)
 
   it("doesn't load the persisted lmdb config if a declarative config is set on reload", function()
     assert(helpers.reload_kong("off", "reload --prefix " .. helpers.test_conf.prefix, {
       database = "off",
       declarative_config = yaml_file,
     }))
-    local res
-    helpers.wait_until(function()
-      local proxy_client = assert(helpers.proxy_client())
-      res = assert(proxy_client:get("/test", { headers = { host = "example1.dev" } }))
-      proxy_client:close()
-      return res.status == 401
-    end)
 
-    local proxy_client = assert(helpers.proxy_client())
```

Comment:
Is there a reason to also include this part in "eventually"? Is 401 not enough to indicate that the new config takes effect?

Reply:
401 just means that the new config is loaded, but not indicate that the old config has been cleared.

Reply:
Of course, we can move the 404 test outside the `eventually`, its not a big deal.